homeassistant/components/waze_travel_time/config_flow.py

```diff
@@ -0,0 +1,35 @@
+"""Config flow for Waze Travel Time integration."""
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME
+from homeassistant.util import slugify
+
+from .const import CONF_DESTINATION, CONF_ORIGIN, DEFAULT_NAME, DOMAIN, WAZE_SCHEMA
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Waze Travel Time."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        if user_input is not None:
+            await self.async_set_unique_id(
+                slugify(
+                    f"{DOMAIN}_{user_input[CONF_ORIGIN]}_{user_input[CONF_DESTINATION]}"
+                )
+            )
+            self._abort_if_unique_id_configured()
```

Comment:
Can we validate that waze can lookup the origin and destination before creating the entry?

Reply:
I think that would be best achieved in a separate PR. As far as I can tell, the component has never done such a validation, so this would be a net new feature.

Reply:
You could create object and call `waze_data.update()` to verify that it works.

But that probably doesn't matter for imports, only ones created in the UI